source/processes/ns_hpv.cpp

```diff
@@ -208,6 +209,23 @@ struct NsService : SessionToPort {
             auto description = fmt::format( "LaunchTitle, title_id={:#x}, flags={:#x}", title_id, flags);
             Session::OnRequest(hypervisor, thread, session, description);
         });
+        
+        dispatcher.DecodeRequest<NS::CardUpdateInitialize>([&](auto& response, uint32_t shared_mem_size, Handle /* shared_mem */) {
+            auto description = fmt::format( "CardUpdateInitialize, shared_mem_size={:#x}", shared_mem_size);
+            Session::OnRequest(hypervisor, thread, session, description);
+            
+            auto gamecard = thread.GetOS().setup.gamecard.get();
+            assert(gamecard);
+            // Disable card update process for 3DSX files, CXI files, and CCI images without an update partition
+            if (!gamecard->HasPartition(Loader::NCSDPartitionId::UpdateData)) {
+                thread.WriteTLS(0x80, 0xFFFE0000); // make NS think a bogus command is sent
+                
+                response.OnResponse([=](Hypervisor& hv, Thread& thread, Result result) {
+                    thread.WriteTLS(0x80, NS::CardUpdateInitialize::response_header);
+                    thread.WriteTLS(0x84, 0xc821180b); // "No card update required"
+                });
+            }
```

Comment:
Since HPV is normally used for introspection only, let's be even more explicit about what's happening here:
```suggestion
            // Disable card update process for 3DSX files, CXI files, and CCI images without an update partition
            if (!gamecard->HasPartition(Loader::NCSDPartitionId::UpdateData)) {
                // Change the request command ID to 0xFFFE0000 to make NS ignore the command.
                thread.WriteTLS(0x80, 0xFFFE0000);

                // Override the response data to indicate that no update is required
                response.OnResponse([=](Hypervisor&, Thread& thread, Result) {
                    thread.WriteTLS(0x80, NS::CardUpdateInitialize::response_header);
                    thread.WriteTLS(0x84, 0xc821180b); // "No card update required"
                });
            }
```